### PR TITLE
Simple logging (revised)

### PR DESCRIPTION
**Pull request by eddowh**
**Change Summaries**

- Dictionary logging configuration in `okqconfig`
- Logger setup used per module in `openkongqi.utils`
- Logging only exists for `openkongqi.source.base` and `openkongqi.source.pm25in`

### CODE REVIEW

**Comment by hrbonz** [comment]
A few simple things to do:
* call the settings entry `LOGGING`
* load the logging configuration in the default conf instead of the user conf, that creates an acceptable default behaviour.

We need to create a common logger for the app, it's more efficient than creating a new instance everywhere. Either create a logger instance at the end of the conf: `logger = xxx` so I can import logger from `openkongqi.conf` **OR** create `openkongqi.log` and create a logger instance there then import `openkongqi.log` in `openkongqi.conf` to raise exceptions as early as possible.

**Comment by eddowh** [comment]
Removed `pythonify` changes in this branch; will open a new separate issue eventually

**Comment by hrbonz** [comment]
Let's take a sensible approach:
* remove error log file
* default log file name should be `openkongqi.log`
* give the choice to change the log file name via settings: `LOGFILE`
* give an option to run in debug mode: `DEBUG` is a boolean, default is `False`
* put the whole logging dict in a `DEFAULT_LOGGING` variable in `openkongqi.conf`
* create a settings entry called `LOGGING` that equals `DEFAULT_LOGGING`
* create a `load_logging` that takes the settings structure:
    * replace `%LOGFILE% with the proper settings value.
    * if `DEBUG` is `True` then add `stdout` logging for `DEBUG` level.

File: openkongqi/conf.py [hunk 1]
```diff
@@ -3,19 +3,61 @@
...
+            'class': 'logging.StreamHandler',
+            'formatter': 'brief',
+            'level': 'DEBUG',
+            'stream': 'ext://sys.stdout'
+        },
+        'info_file_handler': {
+            'backupCount': 20,
+            'class': 'logging.handlers.RotatingFileHandler',
+            'encoding': 'utf8',
+            'filename': '%LOGFILE%',
```
Comment (hrbonz) [inline openkongqi/conf.py]: comment this entry and point to the `global_settings['LOGFILE']` entry

File: openkongqi/conf.py [hunk 2]
```diff
@@ -33,6 +78,18 @@
...
 }
 
+
+def load_logging():
+    # set log file name
+    settings['LOGGING']['handlers']['info_file_handler']['filename'] = \
+        settings['LOGFILE']
+    # log to console if debug configured
+    if settings['DEBUG']:
+        settings['LOGGING']['loggers']['']['handlers'] += ['console']
```
Comment (hrbonz) [inline openkongqi/conf.py]: For style sake, always add a comma in a list with only 1 item (tuples are invalid if you don't write `('foo', )`)
In that case it's probably easier to simply use [`list.append()`]()

**Pull request merged**